.github/workflows/terraform-eks.yml

```diff
@@ -0,0 +1,128 @@
+name: "Terraform"
+
+on:
+  workflow_call:
+    inputs:
+      terraform_root_path:
+        description: the path to the root module to apply
+        type: string
+        required: false
+        default: .
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      BOT_GITHUB_TOKEN:
+        required: false
+      DOCKER_USERNAME:
+        required: false
+      DOCKER_PASSWORD:
+        required: false
+      RANCHER2_ACCESS_KEY:
+        required: false
+      RANCHER2_SECRET_KEY:
+        required: false
+
+jobs:
+  compute_basic_vars:
+    name: compute basic variables
+    runs-on: ubuntu-latest
+    steps:
+      - name: Evaluate branch name
+        id: basic_vars
+        run: |
+          BRANCH_NAME=${{ github.base_ref || github.ref_name }}
+          BRANCH_LOWER=${BRANCH_NAME,,}
+          if [ "$BRANCH_LOWER" == "main" ]; then
+            environment="production"
+          elif [ "$BRANCH_LOWER" == "develop" ]; then
+            environment="develop"
+          else
+            echo "$BRANCH_LOWER doesn't have a matching environment"
+            exit 1
+          fi
+          echo "environment_name=${environment}" >> $GITHUB_OUTPUT
+    outputs:
+      environment_name: ${{ steps.basic_vars.outputs.environment_name }}
+
+  terraform:
+    runs-on: ubuntu-latest
+    needs:
+      - compute_basic_vars
+    environment: ${{ needs.compute_basic_vars.outputs.environment_name }}
+    concurrency:
+      group: ${{ needs.compute_basic_vars.outputs.environment_name }}
+      cancel-in-progress: false
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_DEFAULT_REGION: ${{ vars.AWS_DEFAULT_REGION }}
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      RANCHER2_ACCESS_KEY: ${{ secrets.RANCHER2_ACCESS_KEY }}
+      RANCHER2_SECRET_KEY: ${{ secrets.RANCHER2_SECRET_KEY }}
+      RANCHER2_URL: ${{ vars.RANCHER2_URL }}
+      RESOURCE_NAME: ${{ vars.RESOURCE_NAME }}
+      TERRAFORM_HTTP_CREDENTIALS: |
+        github.com/Alfresco=alfresco-build:${{ secrets.BOT_GITHUB_TOKEN }}
+      TERRAFORM_PRE_RUN: |
+        if [ ! -x ./aws/install ]; then
+          curl -sSf https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip -o awscliv2.zip
+          unzip -q awscliv2.zip
+        fi
+        ./aws/install
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Load environment variables from yml
+        uses: Alfresco/alfresco-build-tools/.github/actions/env-load-from-yaml@v5.0.0
```

Comment:
I'm not sure about reusing that action. It uses a dodgy syntax inherited from travis.

Reply:
not really afraid of it, could be improved, but it needs to keep backward compatibility because it's pretty used